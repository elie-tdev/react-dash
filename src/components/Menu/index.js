export { default as MobileMenu } from './MobileMenu';
export { default as MobileMenuItem } from './MobileMenuItem';
