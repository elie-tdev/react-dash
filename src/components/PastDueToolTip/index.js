export { default as PastDueToolTip } from './PastDueToolTip';
