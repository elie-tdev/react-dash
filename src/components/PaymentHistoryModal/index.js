export { default as PaymentHistoryModal } from './PaymentHistoryModal';
