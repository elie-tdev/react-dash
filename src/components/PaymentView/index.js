export { default as PaymentView } from './PaymentView';
export { default as PaymentCurrent } from './PaymentCurrent';
export { default as PaymentPastDue } from './PaymentPastDue';
export { default as ScheduledPayment } from './ScheduledPayment';
export { default as NextPaymentView } from './NextPaymentView';
export { default as EnrollInAutoPay } from './EnrollInAutoPay';
export { default as NextPaymentCard } from './NextPaymentCard';
export { default as PastDueCard } from './PastDueCard';
export { default as AutoPayCard } from './AutoPayCard';
