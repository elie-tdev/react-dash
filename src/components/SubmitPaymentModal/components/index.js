export { default as CustomListItem } from './CustomListItem';
export { default as NewPaymentAccount } from './NewPaymentAccount';
export { default as EditPaymentAccount } from './EditPaymentAccount';
