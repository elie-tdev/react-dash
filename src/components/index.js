export * from './LoanGlance';
export * from './Button';
export * from './Header';
export * from './Footer';
export * from './LabelValueColumn';
export * from './LabelValueRow';
export * from './Loader';
export * from './CenteredScreenContent';
export * from './PaymentView';
export * from './Pagination';
export * from './Screen';
export * from './LoanSelect';
export * from './StatementGlance';
export * from './Menu';
export * from './PaymentHistoryModal';
export * from './PaymentStatus';
export * from './PaymentType';
export * from './SubmitPaymentModal';
export * from './GridContainer';
export * from './PastDueToolTip';
