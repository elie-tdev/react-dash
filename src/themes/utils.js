export const opacity = (hex, opacity) => {
  var result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
  if (!result) {
    throw new Error('Invalid hex', hex);
  }
  return `rgba(${parseInt(result[1], 16)}, ${parseInt(result[2], 16)}, ${parseInt(
    result[3],
    16,
  )}, ${opacity})`;
};

export const gradient = ({ type = 'linear', colors, background }) =>
  `${type}-gradient(${colors.join(', ')}), ${background};`;
