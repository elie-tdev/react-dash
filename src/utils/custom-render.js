import React from 'react';
import { I18nextProvider } from 'react-i18next';
import { render } from '@testing-library/react';
import { CssBaseline } from '@material-ui/core';
import { BrowserRouter as Router } from 'react-router-dom';
import { ThemeProvider } from '@material-ui/core/styles';
import { MockedProvider } from '@apollo/client/testing';

import { MenuContextProvider } from '@/contexts';
import i18n from '@/translations';
import { LightTheme } from '@/themes';

const wrapper = ({ mocks = [], menuContextProviderProps = {} }) => {
  const Wrapper = ({ children }) => {
    return (
      <Router>
        <MockedProvider mocks={mocks} addTypename={false}>
          <ThemeProvider theme={LightTheme}>
            <MenuContextProvider {...menuContextProviderProps}>
              <CssBaseline />
              <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
            </MenuContextProvider>
          </ThemeProvider>
        </MockedProvider>
      </Router>
    );
  };

  return Wrapper;
};

const customRender = (ui, { mocks, menuContextProviderProps, ...restOptions } = {}) =>
  render(ui, { wrapper: wrapper({ mocks, menuContextProviderProps }), ...restOptions });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
