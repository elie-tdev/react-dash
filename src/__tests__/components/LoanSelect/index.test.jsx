import React from 'react';

import { fireEvent, render } from '@/utils/custom-render';
import { LoanSelect } from '@/components';

const loan = {
  Acctrefno: '123456',
  Current_Payoff_Balance: 10000,
  Total_Current_Due_Balance: 500,
};

describe('LoanSelect', () => {
  test('Renders a single loan', () => {
    const { container } = render(<LoanSelect loans={[loan]} />);
    expect(container).toMatchSnapshot();
  });

  test('Renders multiple loans', () => {
    const { container } = render(<LoanSelect loans={[loan, loan]} />);
    expect(container).toMatchSnapshot();
  });

  test('onClick of select', async () => {
    const handleSelect = jest.fn();
    const handleClick = jest.fn();
    const { container, getAllByTestId } = render(
      <LoanSelect
        loans={[loan, loan]}
        selectedLoanNumber="337720"
        onSelect={handleSelect()}
        onClick={handleClick()}
      />,
    );
    const button = await getAllByTestId('SecondaryButton');
    await expect(button).toHaveLength(1);
    await fireEvent.click(button[0]);
    await expect(handleClick).toHaveBeenCalled();
    await expect(container).toMatchSnapshot();
  });
});
