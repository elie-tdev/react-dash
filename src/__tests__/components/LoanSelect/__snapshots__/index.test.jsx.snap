// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoanSelect Renders a single loan 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-36 makeStyles-column-16"
    data-testid="LoanSelectButton"
  />
</div>
`;

exports[`LoanSelect Renders multiple loans 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-72 makeStyles-column-52"
    data-testid="LoanSelectButton"
  >
    <button
      class="MuiButtonBase-root MuiButton-root makeStyles-root-73 makeStyles-root-74 MuiButton-contained MuiButton-disableElevation MuiButton-fullWidth"
      data-testid="SecondaryButton"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        
        <span
          class="MuiButton-endIcon MuiButton-iconSizeMedium"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
            />
          </svg>
        </span>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`LoanSelect onClick of select 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-110 makeStyles-column-90"
    data-testid="LoanSelectButton"
  >
    <button
      class="MuiButtonBase-root MuiButton-root makeStyles-root-111 makeStyles-root-112 MuiButton-outlined MuiButton-disableElevation MuiButton-fullWidth"
      data-testid="SecondaryButton"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        
        <span
          class="MuiButton-endIcon MuiButton-iconSizeMedium"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z"
            />
          </svg>
        </span>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    class="MuiBox-root MuiBox-root-113 makeStyles-column-90 makeStyles-screenPaddingOn-101 makeStyles-switcher-80"
    data-testid="LoanSelectMenu"
  >
    <div
      class="MuiPaper-root MuiCard-root makeStyles-card-84 MuiPaper-elevation15 MuiPaper-rounded"
    >
      <div
        class="MuiBox-root MuiBox-root-114 makeStyles-accountsCardContent-78 makeStyles-column-90"
      >
        <div
          class="MuiBox-root MuiBox-root-123 makeStyles-column-90 makeStyles-accountItem-115 makeStyles-cardBackground-99"
          data-testid="LoanSelectItem"
        >
          <div
            class="MuiBox-root MuiBox-root-124 makeStyles-row-88 makeStyles-topRow-120"
          >
            <p
              class="MuiTypography-root makeStyles-accountNumber-116 makeStyles-accountNumberActive-117 MuiTypography-body1"
            />
            <div
              class="makeStyles-accountRadio-121 makeStyles-accountRadioActive-122"
            />
          </div>
          <div
            class="MuiBox-root MuiBox-root-125 makeStyles-row-88"
          >
            <div
              class="MuiBox-root MuiBox-root-131 makeStyles-container-129"
              data-testid="LabelValueColumn"
            >
              <h6
                class="MuiTypography-root makeStyles-labelStyle-118 MuiTypography-h6"
              >
                Remaining Balance
              </h6>
              <p
                class="MuiTypography-root makeStyles-value-126 makeStyles-value-130 makeStyles-valueSmall-128 makeStyles-valueStyle-119 MuiTypography-body1"
              >
                $10,000.00
              </p>
            </div>
            <div
              class="MuiBox-root MuiBox-root-133 makeStyles-container-129"
              data-testid="LabelValueColumn"
            >
              <h6
                class="MuiTypography-root makeStyles-labelStyle-118 MuiTypography-h6"
              >
                Amount Due
              </h6>
              <p
                class="MuiTypography-root makeStyles-value-126 makeStyles-value-132 makeStyles-valueSmall-128 makeStyles-valueStyle-119 MuiTypography-body1"
              >
                $500.00
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-134 makeStyles-column-90 makeStyles-accountItem-115 makeStyles-cardBackground-99"
          data-testid="LoanSelectItem"
        >
          <div
            class="MuiBox-root MuiBox-root-135 makeStyles-row-88 makeStyles-topRow-120"
          >
            <p
              class="MuiTypography-root makeStyles-accountNumber-116 makeStyles-accountNumberActive-117 MuiTypography-body1"
            />
            <div
              class="makeStyles-accountRadio-121 makeStyles-accountRadioActive-122"
            />
          </div>
          <div
            class="MuiBox-root MuiBox-root-136 makeStyles-row-88"
          >
            <div
              class="MuiBox-root MuiBox-root-138 makeStyles-container-129"
              data-testid="LabelValueColumn"
            >
              <h6
                class="MuiTypography-root makeStyles-labelStyle-118 MuiTypography-h6"
              >
                Remaining Balance
              </h6>
              <p
                class="MuiTypography-root makeStyles-value-126 makeStyles-value-137 makeStyles-valueSmall-128 makeStyles-valueStyle-119 MuiTypography-body1"
              >
                $10,000.00
              </p>
            </div>
            <div
              class="MuiBox-root MuiBox-root-140 makeStyles-container-129"
              data-testid="LabelValueColumn"
            >
              <h6
                class="MuiTypography-root makeStyles-labelStyle-118 MuiTypography-h6"
              >
                Amount Due
              </h6>
              <p
                class="MuiTypography-root makeStyles-value-126 makeStyles-value-139 makeStyles-valueSmall-128 makeStyles-valueStyle-119 MuiTypography-body1"
              >
                $500.00
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
